lib/charms/smtp_integrator/v0/smtp.py

```diff
@@ -244,6 +244,38 @@ def __init__(self, charm: ops.CharmBase, relation_name: str = DEFAULT_RELATION_N
         self.relation_name = relation_name
         self.framework.observe(charm.on[relation_name].relation_changed, self._on_relation_changed)
 
+    def get_relation_data(self) -> SmtpRelationData:
+        """Retrieve the relation data.
+
+        Returns:
+            SmtpRelationData: the relation data.
+        """
+        relation = self.model.get_relation(self.relation_name)
+        assert relation
```

Comment:
Couldn't this case return an exception?
Example: https://github.com/canonical/synapse-operator/blob/main/lib/charms/data_platform_libs/v0/data_interfaces.py#L879

Reply:
Indeed. Thx